src/main/java/uk/gov/pay/adminusers/queue/event/EventMessageHandler.java

```diff
@@ -1,42 +1,122 @@
+    public EventMessageHandler(EventSubscriberQueue eventSubscriberQueue, LedgerService ledgerService, NotificationService notificationService, ServiceFinder serviceFinder, UserDao userDao, ObjectMapper objectMapper) {
         this.eventSubscriberQueue = eventSubscriberQueue;
+        this.ledgerService = ledgerService;
+        this.notificationService = notificationService;
+        this.serviceFinder = serviceFinder;
+        this.userDao = userDao;
+        this.objectMapper = objectMapper;
     }
-    
+
     public void processMessages() throws QueueException {
         List<EventMessage> eventMessages = eventSubscriberQueue.retrieveEvents();
         for (EventMessage message : eventMessages) {
             try {
                 logger.info("Retrieved event queue message with id {} for resource external id {}",
                         message.getQueueMessage().getMessageId(), message.getEvent().getResourceExternalId());
-                
+                if (message.getEvent().getEventType().equalsIgnoreCase(EventType.DISPUTE.name())) {
+                    handleDisputeCreatedMessage(message.getEvent());
+                } else {
+                    logger.warn("Unknown event type: {}", message.getEvent().getEventType());
+                }
                 eventSubscriberQueue.markMessageAsProcessed(message.getQueueMessage());
             } catch (Exception e) {
                 Sentry.captureException(e);
-                logger.warn("Error during handling the event message",
+                logger.warn("An error occurred handling the event message",
                         kv("sqs_message_id", message.getQueueMessage().getMessageId()),
                         kv("resource_external_id", message.getEvent().getResourceExternalId()),
                         kv("error", e.getMessage())
                 );
             }
         }
     }
+
+    private void handleDisputeCreatedMessage(Event disputeCreatedEvent) throws JsonProcessingException {
+        var disputeCreatedDetails = objectMapper.treeToValue(disputeCreatedEvent.getEventData(), DisputeCreatedDetails.class);
+
+        Service service = serviceFinder.byExternalId(disputeCreatedEvent.getServiceId())
+                .orElseThrow(() -> new IllegalArgumentException(format("Service not found [id: %s]", disputeCreatedEvent.getServiceId())));
+        LedgerTransaction transaction = ledgerService.getTransaction(disputeCreatedEvent.getParentResourceExternalId())
+                .orElseThrow(() -> new IllegalArgumentException(format("Transaction not found [id: %s]", disputeCreatedEvent.getParentResourceExternalId())));
+        
+        List<UserEntity> serviceUsers = userDao.findByServiceId(service.getId());
+        List<UserEntity> serviceAdmins = serviceUsers.stream().filter(userEntity -> {
+            var hasAdminRole = userEntity.getRoles().stream().filter(RoleEntity::isAdmin).count();
+            return hasAdminRole > 0;
+        }).collect(Collectors.toList());
```

Comment:
It might be nice if this were in a service somewhere so we're not directly calling the DAO here and it can be unit tested etc. We'd also potentially want to re-use when contacting admins about other things.

Reply:
👍, added to the UserServices class